### PR TITLE
Limit setting active_fe_indices to locally owned cells

### PR DESCRIPTION
**Pull request by bangerth**
This pull request finishes what #4712 started, namely disallow setting the `active_fe_index` on cells that are not locally owned. This requires a bit of adjusting of one place in the code (the first commit) where the primary component of the patch is to move code around -- the actual change is only the line where we used to call `cell->set_active_fe_index()` and now need to access internal data structures.

The first two commits make sure we don't ever call `set_active_fe_index()` on non-locally owned cells. The third commit actually forbids doing this. The fourth commit documents the changed behavior.

In reference to #3511 and #1366. Also #4704.

### CODE REVIEW

File: include/deal.II/hp/dof_handler.h [hunk 1]
```diff
@@ -132,19 +132,26 @@ namespace hp
...
-   * FE indices on cells that are locally owned or that are ghost cells,
+   * FE indices on cells that are locally owned,
    * using a call such as <code>cell-@>set_active_fe_index(...)</code>.
-   * On the other hand, setting the active FE index on artificial cells
-   * is not allowed.
+   * On the other hand, setting the active FE index on ghost
+   * or artificial cells is not allowed.
    *
-   * However, setting the index on ghost cells has no real effect: whenever
+   * On the other hand, ghost cells do acquire the information what element
```
Comment (tjhei) [inline include/deal.II/hp/dof_handler.h]: too many other hands for my taste. ;-)

File: include/deal.II/dofs/dof_accessor.templates.h [hunk 2]
```diff
@@ -3774,6 +3780,17 @@ DoFCellAccessor<DoFHandlerType,level_dof_access>::set_active_fe_index (const uns
           ExcMessage ("You can not set the active_fe_index on a cell that has "
                       "children because no degrees of freedom will be assigned "
                       "to this cell."));
+
+  if (i != DoFHandlerType::default_fe_index)
```
Comment (tjhei) [inline include/deal.II/dofs/dof_accessor.templates.h]: I don't understand this ``if``, can you please add a comment?

**Comment by bangerth** [comment]
For the record, this passes the testsuite with only unrelated test failures (#4809).

**Comment by bangerth** [comment]
@tjhei -- I made the changes and all `hp` tests continue to work.

**Pull request merged**